<launch>

    <!-- for the GUI -->
    <param name="bound_counter_plot_uav_real_pose" type="int" value="10"/>

    <!-- Controller Node -->
<!--     <node pkg="quad_control" name="plotting_node" type="plot_experiment.py"  output="screen"/> -->

    <!-- Default Mission -->
    <param name="MissionDefault" type="str" value="FireflyTrajectoryTracking"/>

    <!-- Default Yaw Controller -->
    <param name="YawControllerDefault" type="str" value="SimpleTrackingYawController"/>

    <!-- Default Trajectory -->
    <param name="TrajectoryDefault" type="str" value="StayAtRest"/>
    <param name="trajectry_offset_x" type="double" value="0.0"/>
    <param name="trajectry_offset_y" type="double" value="0.0"/>
    <param name="trajectry_offset_z" type="double" value="0.0"/>


    <!-- Iris 2 parameters -->
    <param name="natural_frequency_xy" type="double" value="1.0"/>
    <param name="damping_xy" type="double" value="0.7"/>
    <param name="position_saturation_xy" type="double" value="1.0"/>
    <param name="velocity_saturation_xy" type="double" value="1.0"/>
    <param name="integral_gain_xy" type="double" value="0.0"/>
    <param name="bound_integral_xy" type="double" value="0.0"/>


    <param name="natural_frequency_z" type="double" value="1.4"/>
    <param name="damping_z" type="double" value="1.0"/>
    <!-- <param name="integral_gain_z" type="double" value="0.4"/> -->
    <!-- <param name="bound_integral_z" type="double" value="0.0"/> -->
    <param name="integral_gain_z" type="double" value="0.0"/>
    <param name="bound_integral_z" type="double" value="0.5"/>


    <!-- load lifting gains -->
    <param name="proportional_gain_xy_load_lifting" type="double" value="1.0"/>
    <param name="derivative_gain_xy_load_lifting" type="double" value="1.42"/>
    <param name="cable_angle_gain_load_lifting" type="double" value="1.0"/>
    <param name="cable_angular_velocity_gain_xy_load_lifting" type="double" value="0.0"/>


    <!-- Default Yaw Controller -->
    <param name="ControllerDefault" type="str" value="SimplePIDController"/>
    <param name="DIControllerDefault" type="str" value="ComponentWise3DDIC"/>


    <!-- Default Vector Thrust Controller -->
    <param name="VThrustControllerDefault" type="str" value="VThrustQuadrupleController"/>

    <!-- Default Quadruple Integrator Controller -->
    <param name="QuadrupleIntegratorController" type="str" value="LinearQuadrupleIntController"/>

    <!-- Default Controller for Load transportation by single aerial vehicle -->
    <param name="SingleLoadTransportationControllerDefault" type="str" value="SingleLoadTransportController"/>

    <!-- Controller Node -->
    <node pkg="quad_control" name="controller_Iris1" type="cycle_quad_control_mission.py" output="screen"/>

    <!-- Controller parameters -->
    <param name="gravity_ctr" value="9.81"/>
    <param name="mass_quad_ctr" value="1.442"/>
    <param name="Throttle_neutral_ctr" value="1484.0"/>  
    <param name="ACRO_RP_P_ctr" value="4.5"/> 
    <param name="ktt" value="20.0"/> 
    <param name="ktt2" value="30.0"/>
    <!--  SOME USEFUL RULES: kp = wn*wn  AND kv = 2 xsi wn--> 
    <param name="kv" value="1.0"/> 
    <param name="kp" value="1.4"/> 
    <param name="sigma_p" value="0.5"/> 
    <param name="sigma_v" value="0.5"/> 

    <!-- <param name="quadrotor_mass" value="1.442"/> -->
    <param name="quadrotor_mass" value="1.56779"/>
    <param name="total_mass" type="double" value="1.56779"/>
    <!-- <param name="total_mass" type="double" value="1.66779"/> -->
    

    <!-- load lifting: cable length in meters -->
    <param name="cable_length" value="0.6"/> 
    <!-- load lifting: load mass in kg -->
    <param name="load_mass" value="0.1"/>

    <!-- Simulator parameters -->
    <param name="gravity_sim" value="9.81"/>
    <param name="mass_quad_sim" value="1.442"/>
    <param name="Throttle_neutral_sim" value="1484.0"/>  
    <param name="ACRO_RP_P_sim" value="4.5"/> 
    <param name="ktt_inner_loop" value="10.0"/> 

    <!-- GUI -->
    <!-- <node name="rqt_gui" pkg="gui" type="tabbedGUI"/> -->
    <!-- <node name="rqt_gui" pkg="gui" type="tabPlugin"/> -->

    <!-- This runs rqt and we have to select plugin manually -->
    <!-- similar to rosrun rqt_gui rqt_gui -->
    <!-- <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/> -->

    <!-- for the GUI -->
    <param name="bound_counter_plot_uav_real_pose" type="int" value="10"/>
    <!-- <remap from="uav_real_odometry" to="$(arg uav_odometry)"/> -->
    <remap from="uav_real_odometry" to="uav_odometry"/>
    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--standalone  tabbedGUI --args '' 'gazebo'" output="screen"/>

</launch>